### PR TITLE
Update ndk version to NDK 25b

### PR DESCRIPTION
**Pull request by mohantym**
Update NDK version to NDK 25b  as requested in #57970.

### CODE REVIEW

**Review by JunyoungLim — APPROVED** [review]

**Comment by pkgoogle** [comment]
@mohantym reopening as it seems to resolve an open issue, it seems this was approved but is now a little crusty, any way we can push this forward?

**Comment by gbaned** [comment]
Hi @pjpratik Can you please resolve conflicts in this PR? Thank you!

**Comment by pjpratik** [comment]
Hi @gbaned , I have resolved the conflicts. Can you please check?

Thanks.

File: tensorflow/tools/ci_build/Dockerfile.android [hunk 1]
```diff
@@ -29,7 +29,7 @@ RUN mkdir -p ${ANDROID_DEV_HOME}
 ENV ANDROID_SDK_FILENAME tools_r25.2.5-linux.zip
 ENV ANDROID_SDK_URL https://dl.google.com/android/repository/${ANDROID_SDK_FILENAME}
 ENV ANDROID_API_LEVEL 23
-ENV ANDROID_NDK_API_LEVEL 21
+ENV ANDROID_NDK_API_LEVEL 25
```
Comment (terryheo) [inline tensorflow/tools/ci_build/Dockerfile.android:32]: NDK version and NDK version is different. You don't need to update this.

File: tensorflow/tools/ci_build/builds/builds_common.sh [hunk 2]
```diff
@@ -235,7 +235,7 @@ android_sdk_repository(
 android_ndk_repository(
     name="androidndk",
     path="${ANDROID_NDK_HOME}",
-    api_level=21)
+    api_level=25)
```
Comment (terryheo) [inline tensorflow/tools/ci_build/builds/builds_common.sh:238]: ditto

**Comment by gbaned** [comment]
Hi @pjpratik Can you please check @terryheo's comments and keep us posted ? Thank you!

File: tensorflow/tools/ci_build/Dockerfile.android (hunk 1 above)
Comment (pjpratik) [reply tensorflow/tools/ci_build/Dockerfile.android:32]: Hi @terryheo 

I have reverted the changes back. Thanks.

File: tensorflow/tools/ci_build/builds/builds_common.sh (hunk 2 above)
Comment (pjpratik) [reply tensorflow/tools/ci_build/builds/builds_common.sh]: I have reverted the changes back. Thanks.

**Comment by pjpratik** [comment]
Hi @gbaned @terryheo I have made the changes. 

Thanks.

**Comment by gbaned** [comment]
Hi @pjpratik Can you please resolve conflicts? Thank you!

**Comment by terryheo** [comment]
I submitted the following changes to support NDK 25b.

https://github.com/tensorflow/tensorflow/commit/81658fc94bdb7ca0956cfc1a7907f1bd37e36c8a
https://github.com/tensorflow/tensorflow/commit/0caae80adf169e0106946b2d92faad362854f220

This PR is likely no loner needed?

**Comment by gbaned** [comment]
Closing this PR since changes are submitted by @terryheo. Thank you!

**Pull request closed without merging**